### DISCUSSION TITLE
`./configure.sh` crashing on `LLVM_TABLEGEN_EXE not set`

### DISCUSSION THREAD

**Discussion by billhollings** [3Shain/dxmt#45] [Q&A] [answered]
When running `./configure.sh` I'm running into a crash error on the following errors:

```
CMake Error at cmake/modules/TableGen.cmake:163 (if):
  if given arguments:

    "STREQUAL" "llvm-tblgen"

  Unknown arguments specified
Call Stack (most recent call first):
  utils/TableGen/CMakeLists.txt:5 (add_tablegen)

...

CMake Error at cmake/modules/TableGen.cmake:10 (message):
  LLVM_TABLEGEN_EXE not set
Call Stack (most recent call first):
  include/llvm/IR/CMakeLists.txt:2 (tablegen)
```

Probably just an environment issue, but there's not a lot on the net about this, and I'd appreciate some advice and assistance.

**Comment by 3Shain** [comment, marked as answer]
`brew install llvm@15` I think this should work, or remove the `  -DLLVM_TABLEGEN="$LLVM_TABLEGEN" \` line (there are two).

Well I expected an empty string to be effectively a no-op but now it turns out to be not the case… I may update the document or script later.

**Reply by 3Shain** [reply]
by the way you may skip this part https://github.com/3Shain/dxmt/blob/main/configure.sh#L16-L35, they are only required if airconv windows build is enabled.

**Reply by billhollings** [reply]
Thanks. Installing `llvm@15` via `brew` fixed the config issue. And thanks for highlighting the inapplicable Windows build.

**Reply by fspkwon** [reply]
ive fixed this: https://github.com/3Shain/dxmt/discussions/55 but its macports and llvm15 was only part of the problem also has to enable static libcc and export LLVM_TABLEGEN=/opt/local/bin/llvm-tblgen-mp-15
